src/histolab/slide.py

```diff
@@ -274,113 +277,105 @@ def slides_dimensions(self):
 
-        basic_stats = {
-            "no_of_slides": self.total_slides,
-            "max_width": self._max_width_slide,
-            "max_height": self._max_height_slide,
-            "max_size": self._max_size_slide,
-            "min_width": self._min_width_slide,
-            "min_height": self._min_height_slide,
-            "min_size": self._min_size_slide,
-            "avg_width": self._avg_width_slide,
-            "avg_height": self._avg_height_slide,
-            "avg_size": self._avg_size_slide,
-        }
+    @property
+    def slides_stats(self) -> Tuple[dict, matplotlib_figure]:
+        """Retrieve statistic/graphs of wsi files contained in the dataset"""
 
-        t.elapsed_display()
+        basic_stats = self._dimensions_stats
 
         x, y = zip(*[slide.wsi_dimensions for slide in self.slides])
         colors = np.random.rand(self.total_slides)
         sizes = 8 * self.total_slides
 
-        plt.ioff()
-
-        fig, ax = plt.subplots()
-        ax.scatter(x, y, s=sizes, c=colors, alpha=0.7)
-        plt.xlabel("width (pixels)")
-        plt.ylabel("height (pixels)")
-        plt.title("WSI sizes")
-        plt.set_cmap("prism")
-
-        # plt.scatter(x, y, s=sizes, c=colors, alpha=0.7)
-        # plt.xlabel("width (pixels)")
-        # plt.ylabel("height (pixels)")
-        # plt.title("SVS Image Sizes (Labeled with slide numbers)")
-        # plt.set_cmap("prism")
-        # for i in range(num_images):
-        #     snum = i + 1
-        #     plt.annotate(str(snum), (x[i], y[i]))
-        # plt.tight_layout()
-
-        # area = [w * h / 1e6 for (w, h) in slide_stats]
-        # plt.hist(area, bins=64)
-        # plt.xlabel("width x height (M of pixels)")
-        # plt.ylabel("# images")
-        # plt.title("Distribution of image sizes in millions of pixels")
-        # plt.tight_layout()
-
-        # whratio = [w / h for (w, h) in slide_stats]
-        # plt.hist(whratio, bins=64)
-        # plt.xlabel("width to height ratio")
-        # plt.ylabel("# images")
-        # plt.title("Image shapes (width to height)")
-        # plt.tight_layout()
-
-        # hwratio = [h / w for (w, h) in slide_stats]
-        # plt.hist(hwratio, bins=64)
-        # plt.xlabel("height to width ratio")
-        # plt.ylabel("# images")
-        # plt.title("Image shapes (height to width)")
-        # plt.tight_layout()
-        # t.elapsed_display()
-        return basic_stats, fig
+        figure, axs = plt.subplots(ncols=2, nrows=2, constrained_layout=True)
+
+        axs[0, 0].scatter(x, y, s=sizes, c=colors, alpha=0.7, cmap="prism")
+        axs[0, 0].set_title("SVS Image Sizes (Labeled with slide name)")
+        axs[0, 0].set_xlabel("width (pixels)")
+        axs[0, 0].set_ylabel("height (pixels)")
+        for i, s in enumerate(self.slides):
+            axs[0, 1].annotate(s.wsi_name, (x[i], y[i]))
+
+        area = [w * h / 1e6 for (w, h) in self._wsi_dimensions_list]
```

Comment:
here to change accordingly with the choice above (` _wsi_dimensions_list` vs `_slides_dimensions`)

Reply:
I'm not sure that this is the best choice but slides_dimensions is used for the stats and it's more easy to use to get avg, max etc...
Furthermore wsi_dimensions_list it's just a list of tuple of ints and so more fast than accessing to a dict key each time you need that values. So for now I would keep it separately til we'll find a more pythonic implementation of both

Reply:
???

Reply:
Since we are keeping them both it's ok